Origin: https://git.denx.de/?p=u-boot/u-boot-riscv.git;a=commitdiff;h=081c660201f7567c4fc000e4d758b5ab9f0e8db4
From 081c660201f7567c4fc000e4d758b5ab9f0e8db4 Mon Sep 17 00:00:00 2001
From: David Abdurachmanov <david.abdurachmanov@gmail.com>
Date: Tue, 9 Apr 2019 12:42:47 +0200
Subject: [PATCH] riscv: qemu-riscv.h: define CONFIG_PREBOOT (enables extlinux)

- Set fdt_addr variable, which is needed for extlinux to find FDT.
  Otherwise booting kernel using extlinux results in missing FDT.

- Also run fdt addr with FDT address so that fdt commands would
  work out of the box in U-Boot prompt.

This is successfully used by Fedora/RISCV with 5.1-rc3+ kernel using
OpenSBI -> U-Boot (S-mode) [extlinux] -> Kernel setup.

Signed-off-by: David Abdurachmanov <david.abdurachmanov@gmail.com>
Reviewed-by: Anup Patel <anup.patel@wdc.com>
Reviewed-by: Lukas Auer <lukas.auer@aisec.fraunhofer.de>
Reviewed-by: Bin Meng <bmeng.cn@gmail.com>
---
 include/configs/qemu-riscv.h | 4 ++++
 1 file changed, 4 insertions(+)

Index: u-boot/include/configs/qemu-riscv.h
===================================================================
--- u-boot.orig/include/configs/qemu-riscv.h
+++ u-boot/include/configs/qemu-riscv.h
@@ -48,4 +48,8 @@
 	"ramdisk_addr_r=0x82300000\0" \
 	BOOTENV
 
+#define CONFIG_PREBOOT \
+	"setenv fdt_addr ${fdtcontroladdr};" \
+	"fdt addr ${fdtcontroladdr};"
+
 #endif /* __CONFIG_H */
